plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -775,6 +775,15 @@ public Object[][] targetAndSourceWithDifferentPartitioning()
         };
     }
 
+    // Overriding this test because though delta lake has behavior SUPPORTS_MERGE and
+    // SUPPORTS_NON_NULL_CONSTRAINT, it does not allow update of not null columns
```

Comment:
Shouldn't we be able to remove the check in `DeltaLakeMetadata.beginMerge()` as part of this change? My understanding is that we only added that safety check because we didn't have this non-null handling.

Reply:
Yes, the test for assigning null to a non-null column passes when I remove the check in `beginMerge()`.  Both that check and the delta-lake test removed.